src/mono/wasm/runtime/startup.ts

```diff
@@ -471,6 +472,17 @@ export function bindings_lazy_init(): void {
 
     _create_primitive_converters();
 
+    if (!ENVIRONMENT_IS_PTHREAD) {
+        // Install our sync context so that async continuations will migrate back to the main thread automatically
+        const sync_context_class = cwraps.mono_wasm_assembly_find_class(
+                binding_module, "System.Runtime.InteropServices.JavaScript", "JSSynchronizationContext"
+            ), sync_context_install_method = cwraps.mono_wasm_assembly_find_method(
+                sync_context_class, "InstallSynchronizationContext", 1
```

Comment:
Do we have to call `InstallSynchronizationContext` from JS?  Can it be done from [`StartCallback`](https://github.com/dotnet/runtime/blob/028e556926ad4b03afba9ac1d616cfd3538d36cf/src/mono/System.Private.CoreLib/src/System/Threading/Thread.Mono.cs#L229-L237) ?

But note StartCallback won't be called for:
1. The main thread
2. external (ie native C) pthreads that attach to the runtime

Reply:
We need to install it on the main thread. External threads won't matter even if we decide to install it on the managed worker threads, though.

Reply:
yea I misunderstood what the code was doing.  this seems like a reasonable place to install the sync context, then